DQMServices/Core/interface/locking_ptr.h

```diff
@@ -0,0 +1,136 @@
+#ifndef locking_ptr_h
+#define locking_ptr_h
+
+#include <memory>
+
+namespace detail {
+
+  class lock_base
+  {
+  public:
+    lock_base() = default;
+    virtual ~lock_base() = default;
+
+    virtual void lock() const = 0;
+    virtual void unlock() const = 0;
+  };
+
+  template <typename L>
+  class lock_impl : public lock_base
```

Comment:
Sigh.

Client code needs to know the full type of `locking_ptr<...>` .

There are three possibilities as far as I can tell:
  - the lock type is hard coded
  - the lock type is a template parameter
  - the lock type is hidden via type erasure

I picked the last, because the lock type used by `MonitorElement` may change.